Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -0,0 +1,184 @@
+            return GetWorldTransformURDF(link->getParent(), t);
+        }
+        return t;
+    }
+
+    AZStd::unordered_map<AZStd::string, urdf::LinkSharedPtr> Utils::GetAllLinks(const std::vector<urdf::LinkSharedPtr>& childLinks)
+    {
+        AZStd::unordered_map<AZStd::string, urdf::LinkSharedPtr> pointers;
+        std::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
+            [&](const std::vector<urdf::LinkSharedPtr>& child_links) -> void
+        {
+            for (const urdf::LinkSharedPtr& child_link : child_links)
+            {
+                AZStd::string link_name(child_link->name.c_str(), child_link->name.size());
+                pointers[link_name] = child_link;
+                link_visitor(child_link->child_links);
+            }
+        };
+        link_visitor(childLinks);
+        return pointers;
+    }
+
+    AZStd::unordered_map<AZStd::string, urdf::JointSharedPtr> Utils::GetAllJoints(const std::vector<urdf::LinkSharedPtr>& childLinks)
+    {
+        AZStd::unordered_map<AZStd::string, urdf::JointSharedPtr> joints;
+        std::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
+            [&](const std::vector<urdf::LinkSharedPtr>& child_links) -> void
+        {
+            for (auto child_link : child_links)
+            {
+                const auto& joint = child_link->parent_joint;
+                AZStd::string joint_name(joint->name.c_str(), joint->name.size());
+                joints[joint_name] = joint;
+                link_visitor(child_link->child_links);
+            }
+        };
+        link_visitor(childLinks);
+        return joints;
+    }
+
+    AZStd::unordered_set<AZStd::string> Utils::GetMeshesFilenames(const urdf::LinkConstSharedPtr& rootLink, bool visual, bool colliders)
+    {
+        AZStd::unordered_set<AZStd::string> filenames;
+        const auto addFilenameFromGeometry = [&filenames](const urdf::GeometrySharedPtr& geometry)
+        {
+            if (geometry->type == urdf::Geometry::MESH)
+            {
+                auto pMesh = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+                if (pMesh)
+                {
+                    filenames.insert(AZStd::string(pMesh->filename.c_str(), pMesh->filename.size()));
+                }
+            }
+        };
+
+        const auto processLink = [&addFilenameFromGeometry, visual, colliders](const urdf::Link& link)
+        {
+            if (visual)
+            {
+                for (auto& p : link.visual_array)
+                {
+                    addFilenameFromGeometry(p->geometry);
+                }
+            }
+            if (colliders)
+            {
+                for (auto& p : link.collision_array)
+                {
+                    addFilenameFromGeometry(p->geometry);
+                }
+            }
+        };
+
+        std::function<void(const std::vector<urdf::LinkConstSharedPtr>&)> linkVisitor =
```

Comment:
repeat: AZStd::function can be used here.
```suggestion
        AZStd::function<void(const std::vector<urdf::LinkConstSharedPtr>&)> linkVisitor =
```